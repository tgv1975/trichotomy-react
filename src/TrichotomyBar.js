import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import trichotomy from 'trichotomy/dist/trichotomy.min.css';

export default class TrichotomyBar extends PureComponent {
    static defaultProps = {
        tag: 'div',
        right: false,
    }

    static propTypes = {
        tag: PropTypes.string,
        right: PropTypes.bool,
    }

    render() {
        const {
            children,
            className,
            right,
            tag,
        } = this.props;

        const Tag = tag;

        let cssClass = className || '';

        if (right) {
            cssClass = `${cssClass} ${trichotomy.right}`.trim();
        }

        return (
            <Tag className={cssClass}>
                {children}
            </Tag>
        );
    }
}
