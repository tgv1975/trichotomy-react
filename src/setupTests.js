jest.mock('trichotomy/dist/trichotomy.min.css', () => (
    {
        trichotomy: 'trichotomy',
        vertical: 'vertical',
        horizontal: 'horizontal',
        right: 'right',
        centerpiece: 'centerpiece',
    }
));
