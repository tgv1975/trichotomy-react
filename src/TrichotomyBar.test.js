import React from 'react';
import renderer from 'react-test-renderer';
import './setupTests';

import TrichotomyBar from './TrichotomyBar';

describe('Trichotomy', () => {
    it('Trichotomy is truthy', () => {
        expect(TrichotomyBar).toBeTruthy();
    });

    test('Renders TrichotomyBar container <DIV> element (default)', () => {
        const component = renderer.create(
            <TrichotomyBar />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('div');
    });

    test('Renders TrichotomyBar container <ASIDE> element (custom)', () => {
        const component = renderer.create(
            <TrichotomyBar tag="aside" />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('aside');
    });

    test('Renders TrichotomyBar container element with "right" CSS class if set', () => {
        const component = renderer.create(
            <TrichotomyBar right />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toEqual('right');
    });

    test('Renders TrichotomyBar container element with "someCustomClass right" CSS classes if set', () => {
        const component = renderer.create(
            <TrichotomyBar right className="someCustomClass" />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toEqual('someCustomClass right');
    });

    test('Renders children into TrichotomyBar container element ', () => {
        const component = renderer.create(
            <TrichotomyBar>
                <div>First child</div>
                <div>Second child</div>
                <div>Third child</div>
            </TrichotomyBar>,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree.children).toBeTruthy();
    });
});
