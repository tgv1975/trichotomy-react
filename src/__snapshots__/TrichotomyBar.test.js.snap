// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trichotomy Renders TrichotomyBar container <ASIDE> element (custom) 1`] = `
<aside
  className=""
/>
`;

exports[`Trichotomy Renders TrichotomyBar container <DIV> element (default) 1`] = `
<div
  className=""
/>
`;

exports[`Trichotomy Renders TrichotomyBar container element with "right" CSS class if set 1`] = `
<div
  className="right"
/>
`;

exports[`Trichotomy Renders TrichotomyBar container element with "someCustomClass right" CSS classes if set 1`] = `
<div
  className="someCustomClass right"
/>
`;

exports[`Trichotomy Renders children into TrichotomyBar container element  1`] = `
<div
  className=""
>
  <div>
    First child
  </div>
  <div>
    Second child
  </div>
  <div>
    Third child
  </div>
</div>
`;
